content/updates/terraform-null-label-advanced.md

```diff
@@ -0,0 +1,343 @@
+ context = module.this.context
+}
+
+data "aws_iam_policy_document" "assume_role_policy" {
+ # not included for brevity...
+ statement { }
+}
+
+data "aws_iam_policy_document" "permission_boundary" {
+ # not included for brevity...
+ statement { }
+}
+
+resource "aws_iam_policy" "permission_boundary" {
+ name        = module.permission_boundary_label.id
+ description = "Permission boundary for this custom role."
+ policy      = data.aws_iam_policy_document.permission_boundary.json
+ tags        = module.permission_boundary_label.tags
+}
+
+resource "aws_iam_role" "instance_role" {
+ name                 = module.this.id
+ assume_role_policy   = data.aws_iam_policy_document.assume_role_policy.json
+ permissions_boundary = aws_iam_policy.permission_boundary.arn
+ tags                 = module.this.tags
+}
+
+data "aws_iam_policy_document" "custom_policy" {
+ # not included for brevity...
+ statement { }
+}
+
+resource "aws_iam_policy" "custom_policy" {
+ name        = module.custom_policy_label.id
+ description = "Custom policy for this role."
+ policy      = data.aws_iam_policy_document.custom_policy.json
+ tags        = module.custom_policy_label.tags
+}
+
+resource "aws_iam_role_policy_attachment" "custom_policy" {
+ role       = aws_iam_role.instance_role.name
+ policy_arn = aws_iam_policy.custom_policy.arn
+}
+```
+
+We can source it in the root module like this:
+
+```hcl
+# terraform/main.tf
+module "iam" {
+  source = "./iam-module"
+
+  # This makes the same label values defined in the root module available in the child module due to context.tf
+  context = module.this.context
+}
+```
+
+Since the IAM module has its own `context.tf` file at `terraform/iam-module/context.tf`, it can accept the `context` argument and use the same inputs and labels that are defined in the root module. Although this was an overly simplified example, more common IAM configurations often involve many roles and policies. With `context.tf`, it’s much easier to ensure consistent naming and tagging across IAM resources, which is one of the most common pain points in AWS environments for inconsistency and sprawl.
+
+What if we wanted to use an external module, and wanted to override a couple of values? This is no problem if it implements `context.tf`:
+
+```hcl
+module "rds_cluster_aurora_postgres" {
+  source  = "cloudposse/rds-cluster/aws"
+  version = "1.7.0"
+
+  name = "oltpdb"
+
+  engine      = "aurora-postgresql"
+  cluster_family  = "aurora-postgresql9.6"
+  cluster_size= 2
+
+  admin_user  = "admin1"
+  <!-- trunk-ignore(gitleaks/hashicorp-tf-password) -->
+  admin_password  = "Test123456789"
+  db_name     = "dbname"
+  db_port     = 5432
+  instance_type   = "db.r4.large"
+  vpc_id      = aws_vpc.public.id
+  security_groups = [ aws_security_group.db_security_group.id ]
+  subnets     = [ aws_subnet.public_subnet.id ]
+  zone_id     = "Zxxxxxxxx"
+
+  context = module.this.context
+}
```

Comment:
```suggestion
module "rds_cluster_aurora_postgres" {
  source         = "cloudposse/rds-cluster/aws"
  version        = "1.7.0"
  name           = "oltpdb"
  engine         = "aurora-postgresql"
  cluster_family = "aurora-postgresql9.6"
  cluster_size   = 2
  admin_user     = "admin1"
  # trunk-ignore(gitleaks/hashicorp-tf-password)
  admin_password  = "Test123456789"
  db_name         = "dbname"
  db_port         = 5432
  instance_type   = "db.r4.large"
  vpc_id          = aws_vpc.public.id
  security_groups = [aws_security_group.db_security_group.id]
  subnets         = [aws_subnet.public_subnet.id]
  zone_id         = "Zxxxxxxxx"
  context         = module.this.context
}
```

Reply:
I gotta see if this shows up in the preview or not. We definitely want it to not show up.

Reply:
Changed to passing a `var` here so this is no longer necessary.